src/core/thread/mesh_forwarder.cpp

```diff
@@ -1415,15 +1415,18 @@ void MeshForwarder::HandleReceivedFrame(Mac::Frame &aFrame, ThreadError aError)
     switch (aFrame.GetType())
     {
     case Mac::Frame::kFcfFrameData:
-        if (reinterpret_cast<Lowpan::MeshHeader *>(payload)->IsMeshHeader())
+        if (payloadLength >= sizeof(Lowpan::MeshHeader) &&
+            reinterpret_cast<Lowpan::MeshHeader *>(payload)->IsMeshHeader())
         {
             HandleMesh(payload, payloadLength, messageInfo);
         }
-        else if (reinterpret_cast<Lowpan::FragmentHeader *>(payload)->IsFragmentHeader())
+        else if (payloadLength >= sizeof(Lowpan::FragmentHeader) &&
+                 reinterpret_cast<Lowpan::FragmentHeader *>(payload)->IsFragmentHeader())
         {
             HandleFragment(payload, payloadLength, macSource, macDest, messageInfo);
         }
-        else if (Lowpan::Lowpan::IsLowpanHc(payload))
+        else if (payloadLength >= 1 &&
+                 Lowpan::Lowpan::IsLowpanHc(payload))
         {
             HandleLowpanHC(payload, payloadLength, macSource, macDest, messageInfo);
         }
```

Comment:
Adding hints for static checks sounds interesting.  One classic way to use fuzzing is to enable AddressSanitizer in the build step (per #607), and try to have the random inputs trigger the extra runtime checks such as buffer and stack overflows.  What is the mechanism your fuzzer is using to find failures?